tool/tsh/tshconfig_test.go

```diff
@@ -207,3 +207,56 @@ func TestTshConfigMerge(t *testing.T) {
 		})
 	}
 }
+
+// TestProxyTemplates verifies proxy templates matching functionality.
+func TestProxyTemplates(t *testing.T) {
+	tshConfig := &TshConfig{
+		ProxyTemplates: ProxyTemplates{
+			{
+				Template: `^(.+)\.(us.example.com):(.+)$`,
+				Proxy:    "$2:443",
+				Host:     "$1:$3",
+			},
+			{
+				Template: `^(.+)\.(eu.example.com):(.+)$`,
+				Proxy:    "$2:3080",
```

Comment:
Can we test some tricker replacements? Something with numbers or letters next to the replacement maybe so you need braces, `${1}1-x${2}.example.com:3080` or something